### PR TITLE
Feature/backend structure

### PR DESCRIPTION
**Pull request by Mathilde-Frere**
1- Configure server to fit with websocket usage.

2-Add back-end API about:
- Thruster
- Radio

3-Fix some minor bug on HMI part

### CODE REVIEW

File: src/backend/services/api-radio.service.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    if(typeof(radioState.frequency) !== 'number') {
+      return res.status(StatusCode.ClientErrorBadRequest).send({ msg: 'Radio frequency is not of correct type.' });
+    }
+    if(radioState.frequency < instance._MIN_FREQUENCY || radioState.frequency > instance._MAX_FREQUENCY) {
+      return res.status(StatusCode.ClientErrorBadRequest).send({ msg: `Radio frequency is not in the correct range: [${instance._MIN_FREQUENCY} - ${instance._MAX_FREQUENCY}].` });
+    }
+
+    if(radioState.powerOn !== instance._fakeRadioData.powerOn) {
```
Comment (PierreSachot) [inline src/backend/services/api-radio.service.ts:158]: Here you can only update 1 value at a time. If you change the power status and frequency, only power status is updated. 

Keep it like this, if the futur employe correct this we would know that he is the one ahah

File: src/backend/services/api-thruster.service.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+   * @param { Request } _req - The Request object from Express
+   * @param { Response } res - The Response object from Express
+   * @returns { Promise<Response<any, Record<string, any>>> } The Express object response
+   */
+  @APIThrusterService._checkInstantiated
+  private static async _getThrusterListState(_req: Request, res: Response): Promise<Response<any, Record<string, any>>> {
+    const instance = APIThrusterService.Instance;
+    const thrusterListState = instance.thrusterList;
+
+    return res.status(StatusCode.SuccessOK).send({ thrusterListState });
```
Comment (PierreSachot) [inline src/backend/services/api-thruster.service.ts:188]: Same as before, I would do a JSON.stringify(thrusterListState)

File: src/frontend/app/components/thruster-list-container/thruster-list-container.component.html [hunk 3]
```diff
@@ -2,13 +2,7 @@
   <div>
     <h1>Thrusters</h1>
   </div>
-  <div *ngIf="thrusterList.length !== 0; else error">
+  <div>
```
Comment (PierreSachot) [inline src/frontend/app/components/thruster-list-container/thruster-list-container.component.html:5]: Ahah, Too much help ? :p

**Review by PierreSachot — CHANGES_REQUESTED** [review]
Really good job, thanks a lot! Really nice the random behavior and everything! 

Hope we will find the person we need!

File: src/backend/services/api-radio.service.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+   * @param { Request } _req - The Request object from Express
+   * @param { Response } res - The Response object from Express
+   * @returns { Promise<Response<any, Record<string, any>>> } The Express object response
+   */
+  @APIRadioService._checkInstantiated
+  private static async _getRadioState(_req: Request, res: Response): Promise<Response<any, Record<string, any>>> {
+    const instance = APIRadioService.Instance;
+    const radioState = instance.radio;
+
+    return res.status(StatusCode.SuccessOK).send({ radioState });
```
[the remark this answers is not in the document]
Comment (Mathilde-Frere) [reply src/backend/services/api-radio.service.ts]: It has been chosen to return : `return res.status(StatusCode.SuccessOK).send(radioState);`

File: src/backend/services/api-thruster.service.ts (hunk 2 above)
Comment (Mathilde-Frere) [reply src/backend/services/api-thruster.service.ts]: It has been chosen to return : `return res.status(StatusCode.SuccessOK).send(thrusterListState);`

File: README.md [hunk 5]
```diff
@@ -111,6 +111,7 @@ Your mission is to develop a cutting-edge rocket that will revolutionize space t
 - Document the methods implemented.
 - Commit your changes regularly.
 - Use Angular Material library for toggles, progress bars and the slider (these modules are already available on the app).
+- Data must be updated in real time across multiple open browser windows (websocket network protocol).
```
Comment (PierreSachot) [inline README.md:114]: Great!

**Review by PierreSachot — APPROVED** [review]
Good job! Thanks a lot for your hard work!

We are ready to hire!

**Pull request merged**